src/Crypto/WebAuthn/Operations/Attestation/AndroidSafetyNet.hs

```diff
@@ -0,0 +1,201 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Crypto.WebAuthn.Operations.Attestation.AndroidSafetyNet
+  ( format,
+    Format (..),
+    DecodingError (..),
+    Statement (..),
+    VerificationError (..),
+  )
+where
+
+import Codec.CBOR.Term (Term (TBytes, TString))
+import qualified Codec.CBOR.Term as CBOR
+import Control.Lens ((^.), (^?))
+import Control.Lens.Combinators (_Just)
+import Control.Monad (unless, when)
+import Control.Monad.Except (MonadError, runExcept, throwError)
+import qualified Crypto.Hash as Hash
+import qualified Crypto.JOSE as JOSE
+import qualified Crypto.JOSE.Types as X509
+import qualified Crypto.JWT as JOSE
+import qualified Crypto.WebAuthn.Model as M
+import qualified Data.ASN1.Types.String as X509
+import Data.Aeson ((.=))
+import qualified Data.Aeson as Aeson
+import Data.Bifunctor (Bifunctor (first))
+import qualified Data.ByteArray as BA
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Base64 as Base64
+import qualified Data.ByteString.Char8 as BSC
+import qualified Data.ByteString.Lazy as LB
+import Data.HashMap.Lazy (HashMap, (!?))
+import qualified Data.Hourglass as HG
+import qualified Data.List.NonEmpty as NE
+import qualified Data.Text as Text
+import qualified Data.X509 as X509
+import qualified Data.X509.Validation as X509
+import GHC.Exception (Exception)
+import GHC.Generics (Generic)
+
+data Format = Format
+
+instance Show Format where
+  show = Text.unpack . M.asfIdentifier
+
+-- | [(spec)](https://developer.android.com/training/safetynet/attestation.html#compat-check-response)
+data Response = Response
+  { timestampMs :: Integer,
+    nonce :: String,
+    apkPackageName :: String,
+    apkCertificateDigestSha256 :: [String], -- [Base 64 encoded SHA256 hash]
+    ctsProfileMatch :: Bool,
+    basicIntegrity :: Bool,
```

Comment:
Regarding these values, I think they should be checked to be both true. We can later also make it configurable, which could look like this:
```haskell
data RequiredIntegrity
  = NoIntegrity
  | BasicIntegrity
  | PassedCTS
```

With links to the docs in https://developer.android.com/training/safetynet/attestation.html#compat-check-response